identity-apps-core/apps/recovery-portal/src/main/webapp/username-recovery-channel.jsp

```diff
@@ -0,0 +1,291 @@
+    %>
+        <jsp:include page="extensions/header.jsp"/>
+    <% } else { %>
+        <jsp:include page="includes/header.jsp"/>
+    <% } %>
+
+</head>
+<body class="login-portal layout recovery-layout">
+    <layout:main layoutName="<%= layout %>" layoutFileRelativePath="<%= layoutFileRelativePath %>" data="<%= layoutData %>" >
+        <layout:component componentName="ProductHeader">
+            <%-- product-title --%>
+            <%
+                File productTitleFile = new File(getServletContext().getRealPath("extensions/product-title.jsp"));
+                if (productTitleFile.exists()) {
+            %>
+            <jsp:include page="extensions/product-title.jsp"/>
+            <% } else { %>
+            <jsp:include page="includes/product-title.jsp"/>
+            <% } %>
+        </layout:component>
+        <layout:component componentName="MainSection" >
+            <div class="ui segment">
+                <h2><%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "Recover.username")%></h2>
+                <% if (error) { %>
+                    <div class="ui visible negative message" id="server-error-msg">
+                        <%= IdentityManagementEndpointUtil.i18nBase64(recoveryResourceBundle, errorMsg) %>
+                    </div>
+                <% } %>
+                <div class="ui negative message" id="error-msg" hidden="hidden"></div>
+
+                <%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "Select.the.recovery.option")%>
+
+                <div class="ui divider hidden"></div>
+
+                <div class="segment-form">
+                    <form class="ui large form" method="post" action="verify.do" id="channelSelectionForm">
+                        <%
+                            String callback = IdentityManagementEndpointUtil.getStringValue(request.getAttribute("callback"));
+
+                            if (StringUtils.isBlank(callback)) {
+                                callback = IdentityManagementEndpointUtil.getUserPortalUrl(
+                                        application.getInitParameter(IdentityManagementEndpointConstants.ConfigConstants.USER_PORTAL_URL), tenantDomain);
+                            }
+
+                            if (callback != null) {
+                        %>
+                        <div>
+                            <input type="hidden" name="callback" value="<%=Encode.forHtmlAttribute(callback) %>"/>
+                        </div>
+                        <div>
+                            <input type="hidden" name="sp" value="<%=Encode.forHtmlAttribute(request.getParameter("sp"))%>"/>
+                        </div>
+
+                        <% } %>
+
+                              <div class="field">
+                                <div class="ui radio checkbox">
+                                    <input type="radio" name="usernameRecoveryOption" value="<%=getChannelIdFromChannelName(channels, EMAIL) + ":" + EMAIL%>"
```

Comment:
check if we can encode this too sine we are using it as a value.

Reply:
Fixed by https://github.com/wso2/identity-apps/pull/7005/commits/1602b3faee1134d758e7ef9fd57bc6ad5bdd1db9